api/availability.go

```diff
@@ -67,13 +69,26 @@ func (nc *NetworkChecker) Start(ctx context.Context) error {
 	}
 
 	go func(ctx context.Context) {
-		for range time.NewTicker(nc.checkPeriod).C {
-			nc.updateAvailability(ctx)
+		ticker := time.NewTicker(nc.checkPeriod)
+		stop := false
+		for !stop {
+			select {
+			case <-ticker.C:
+				nc.updateAvailability(ctx)
+			case <-nc.stopped:
+				stop = true
+			}
 		}
+		nc.stopped <- struct{}{}
 	}(ctx)
 	return nil
 }
 
+func (nc *NetworkChecker) Stop() {
+	nc.stopped <- struct{}{}
+	<-nc.stopped
```

Comment:
FYI: Since you are using a non-buffered channel you can avoid making a double check here. When first `<-` unblocks you can be sure that the goroutine is about to terminate.